### ISSUE TITLE
Crash report 6.2: ArithmeticException in ReaderSiteHeaderView.showBlogInfo

### ISSUE THREAD

**Issue by maxme** [wordpress-mobile/WordPress-Android#5568]
It's a java.lang.String.format [bug](https://bugs.openjdk.java.net/browse/JDK-8167567), but it's also very common, we should catch the exception or remove the grouping format.

```
Fatal Exception: java.lang.ArithmeticException: divide by zero
       at java.util.Formatter$FormatSpecifier.localizedMagnitude(Formatter.java:4210)
       at java.util.Formatter$FormatSpecifier.print(Formatter.java:3140)
       at java.util.Formatter$FormatSpecifier.print(Formatter.java:3121)
       at java.util.Formatter$FormatSpecifier.printInteger(Formatter.java:2826)
       at java.util.Formatter$FormatSpecifier.print(Formatter.java:2786)
       at java.util.Formatter.format(Formatter.java:2491)
       at java.util.Formatter.format(Formatter.java:2426)
       at java.lang.String.format(String.java:2626)
       at org.wordpress.android.ui.reader.views.ReaderSiteHeaderView.showBlogInfo(ReaderSiteHeaderView.java:132)
       at org.wordpress.android.ui.reader.views.ReaderSiteHeaderView.loadBlogInfo(ReaderSiteHeaderView.java:79)
       at org.wordpress.android.ui.reader.adapters.ReaderPostAdapter.onBindViewHolder(ReaderPostAdapter.java:317)
       at android.support.v7.widget.RecyclerView$Adapter.onBindViewHolder(RecyclerView.java:6308)
       at android.support.v7.widget.RecyclerView$Adapter.bindViewHolder(RecyclerView.java:6341)
       at android.support.v7.widget.RecyclerView$Recycler.tryBindViewHolderByDeadline(RecyclerView.java:5287)
       at android.support.v7.widget.RecyclerView$Recycler.tryGetViewHolderForPositionByDeadline(RecyclerView.java:5550)
       at android.support.v7.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:5392)
       at android.support.v7.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:5388)
       at android.support.v7.widget.LinearLayoutManager$LayoutState.next(LinearLayoutManager.java:2149)
       at android.support.v7.widget.LinearLayoutManager.layoutChunk(LinearLayoutManager.java:1533)
       at android.support.v7.widget.LinearLayoutManager.fill(LinearLayoutManager.java:1496)
       at android.support.v7.widget.LinearLayoutManager.onLayoutChildren(LinearLayoutManager.java:593)
       at android.support.v7.widget.RecyclerView.dispatchLayoutStep2(RecyclerView.java:3535)
       at android.support.v7.widget.RecyclerView.dispatchLayout(RecyclerView.java:3264)
       at android.support.v7.widget.RecyclerView.onLayout(RecyclerView.java:3796)
       at android.view.View.layout(View.java:17535)
       at android.view.ViewGroup.layout(ViewGroup.java:5616)
       at android.support.v4.widget.SwipeRefreshLayout.onLayout(SwipeRefreshLayout.java:636)
       at android.view.View.layout(View.java:17535)
       at android.view.ViewGroup.layout(ViewGroup.java:5616)
       at android.widget.RelativeLayout.onLayout(RelativeLayout.java:1079)
       at android.view.View.layout(View.java:17535)
       at android.view.ViewGroup.layout(ViewGroup.java:5616)
       at android.support.design.widget.HeaderScrollingViewBehavior.layoutChild(HeaderScrollingViewBehavior.java:131)
       at android.support.design.widget.ViewOffsetBehavior.onLayoutChild(ViewOffsetBehavior.java:42)
       at android.support.design.widget.AppBarLayout$ScrollingViewBehavior.onLayoutChild(AppBarLayout.java:1375)
       at android.support.design.widget.CoordinatorLayout.onLayout(CoordinatorLayout.java:870)
       at android.view.View.layout(View.java:17535)
       at android.view.ViewGroup.layout(ViewGroup.java:5616)
       at android.widget.RelativeLayout.onLayout(RelativeLayout.java:1079)
       at android.view.View.layout(View.java:17535)
       at android.view.ViewGroup.layout(ViewGroup.java:5616)
       at android.widget.RelativeLayout.onLayout(RelativeLayout.java:1079)
       at android.view.View.layout(View.java:17535)
       at android.view.ViewGroup.layout(ViewGroup.java:5616)
       at android.support.v4.view.ViewPager.onLayout(ViewPager.java:1795)
       at android.view.View.layout(View.java:17535)
       at android.view.ViewGroup.layout(ViewGroup.java:5616)
       at android.widget.RelativeLayout.onLayout(RelativeLayout.java:1079)
       at android.view.View.layout(View.java:17535)
       at android.view.ViewGroup.layout(ViewGroup.java:5616)
       at android.widget.FrameLayout.layoutChildren(FrameLayout.java:323)
       at android.widget.FrameLayout.onLayout(FrameLayout.java:261)
       at android.view.View.layout(View.java:17535)
       at android.view.ViewGroup.layout(ViewGroup.java:5616)
       at android.widget.FrameLayout.layoutChildren(FrameLayout.java:323)
       at android.widget.FrameLayout.onLayout(FrameLayout.java:261)
       at android.view.View.layout(View.java:17535)
       at android.view.ViewGroup.layout(ViewGroup.java:5616)
       at android.widget.FrameLayout.layoutChildren(FrameLayout.java:323)
       at android.widget.FrameLayout.onLayout(FrameLayout.java:261)
       at android.view.View.layout(View.java:17535)
       at android.view.ViewGroup.layout(ViewGroup.java:5616)
       at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1741)
       at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1585)
       at android.widget.LinearLayout.onLayout(LinearLayout.java:1494)
       at android.view.View.layout(View.java:17535)
       at android.view.ViewGroup.layout(ViewGroup.java:5616)
       at android.widget.FrameLayout.layoutChildren(FrameLayout.java:323)
       at android.widget.FrameLayout.onLayout(FrameLayout.java:261)
       at com.android.internal.policy.DecorView.onLayout(DecorView.java:724)
       at android.view.View.layout(View.java:17535)
       at android.view.ViewGroup.layout(ViewGroup.java:5616)
       at android.view.ViewRootImpl.performLayout(ViewRootImpl.java:2354)
       at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2081)
       at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1258)
       at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:6348)
       at android.view.Choreographer$CallbackRecord.run(Choreographer.java:871)
       at android.view.Choreographer.doCallbacks(Choreographer.java:683)
       at android.view.Choreographer.doFrame(Choreographer.java:619)
       at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:857)
       at android.os.Handler.handleCallback(Handler.java:751)
       at android.os.Handler.dispatchMessage(Handler.java:95)
       at android.os.Looper.loop(Looper.java:154)
       at android.app.ActivityThread.main(ActivityThread.java:6123)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:867)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:757)
```

**Comment by tonyr59h** [comment]
```CUSTOMER SUBMITTED WORKAROUND : 
Eliminating the formatting, e.g. String.format("%d", 1000) 
Specifying locale, e.g. String.format(Locale.US, "Less than %,d", 100000) ```

Per the reported workaround it seems we could specify the Locale for `String.format` to get around the issue.

**Comment by tonyr59h** [comment]
906dabd1819563fc4a6c4c3a54059ab49857983d passes the locale into `String.format` and, just in case that doesn't solve the problem, encloses the problem code in a try/catch block.

**Comment by maxme** [comment]
> Per the reported workaround it seems we could specify the Locale for String.format to get around the issue.

I don't think this will changed anything, this is already what's happening (current locale is used). I think the proposed workaround is to force the use of a certain locale (like Locale.US in this example).

**Comment by tonyr59h** [comment]
Oh, I think you're right, I misread it. I'll revert that and just add the `try/catch` block in that case.

**Comment by mzorz** [comment]
closed via #5604
